devicetypes/zooz/zooz-zen51-dry-contact-relay.src/zooz-zen51-dry-contact-relay.groovy

```diff
@@ -0,0 +1,297 @@
+@Field static int btnHeld = 2
+@Field static List<String> supportedButtonValues = ["pushed","held","pushed_2x","pushed_3x","pushed_4x","pushed_5x"]
+
+@Field static Map configParams = [
+	ledIndicator: [num:1, title:"Led indicator", size:1, defaultVal:1, options:[0:"Disabled", 1:"Enabled"]],
+	autoOff: [num:2, title:"Auto Off Timer", size:2, defaultVal:0, range:"0..65535", desc:"0(disabled), 1..65535(timer unit)"],
+	autoOn: [num:3, title:"Auto On Timer", size:2, defaultVal:0, range:"0..65535", desc:"0(disabled), 1..65535(timer unit)"],
+	timerUnit: [num:10, title:"Timer Unit", size:1, defaultVal:1, options:[1:"Minutes", 2:"Seconds"]],
+	statusAfterPowerFailure: [num:4, title:"On/Off Status After Power Failure", size:1, defaultVal:2, options:[0:"Forced off", 1:"Forced on", 2:"Restore previous state"]],
+	loadControl: [num:6, title:"Load Control", size:1, defaultVal:1, options:[0:"Disable Switch/ Enable Z-Wave", 1:"Enable Switch and Z-Wave", 2:"Disable Switch and Z-Wave"]],
+	switchType: [num:7, title:"Switch Type", size:1, defaultVal:2, options:[0:"Toggle Switch", 1:"Momentary Light Switch", 2:"Toggle Up On/Down Off", 3:"3-way Impulse Control", 4:"Garage Door Mode"]],
+	relayBehavior: [num:9, title:"Relay Type Behavior", size:1, defaultVal:0, options:[0:"Normally Open (NO)", 1:"Normally Closed (NC)"]],
+	impulseDuration: [num:11, title:"Impulse Duration for 3-way", size:1, defaultVal:10, range:"2..200", desc:"2..200 (seconds)"]
+]
+
+metadata {
+	definition (
+		name: "Zooz ZEN51 Dry Contact Relay",
+		namespace: "Zooz",
+		author: "Kevin LaFramboise (@krlaframboise)",
+		ocfDeviceType: "oic.d.light",
+		mnmn: "SmartThingsCommunity",
+		vid: "d4bdecb2-4374-3c96-aceb-24223399fe5f"
+	) {
+		capability "Actuator"
+		capability "Sensor"
+		capability "Switch"
+		capability "Button"
+		capability "Refresh"
+		capability "Health Check"
+		
+		// zw:Ls2a type:1000 mfr:027A prod:0104 model:0201 ver:1.24 zwv:7.15 lib:03 cc:5E,55,9F,6C,22 sec:25,70,85,59,8E,86,72,5A,73,7A,5B,87
+		fingerprint mfr: "027A", prod: "0104", model: "0201", deviceJoinName: "Zooz Switch" // Zooz ZEN51 Dry Contact Relay
+	}
+
+	preferences {
+		configParams.each { name, param ->
+			if (param.options) {
+				input name, "enum",
+					title: param.title,
+					description: "Default: ${param.options[param.defaultVal]}",
+					required: false,
+					displayDuringSetup: false,
+					defaultValue: param.defaultVal,
+					options: param.options
+			} else if (param.range) {
+				input name, "number",
+					title: param.title,
+					description: "${param.desc} - Default: ${param.defaultVal}",
+					required: false,
+					displayDuringSetup: false,
+					defaultValue: param.defaultVal,
+					range: param.range
+			}
+		}
+	}
+}
+
+def installed() {
+	log.debug "installed()..."
+	state.firstConfig = true
+}
+
+def updated() {
+	log.debug "updated()..."
+	initialize()
+	
+	if (!state.firstConfig) {
+		configure()
+	} else {
+		state.firstConfig = false
+	}
+}
+
+void initialize() {
```

Comment:
if you just called this from `installed` and not from `updated`, you wouldn't need to have these if statements

Reply:
The "configure" method has to execute from the "updated" method to sync the setting changes, but the "updated" method gets called multiple times during inclusion and without that workaround I kept getting random issues during inclusion like security failed warnings and the device getting joined without appearing in ST.

I guess it's possible the platform was having issues while I was writing the handler, but this was a relatively clean workaround and it solved the problem...

Reply:
Updated is fine. I'm talking about the stuff you're doing here in initialize: generating initial state and setting the checkinterval

why can't you put it in installed? are you running into the case where the checkInterval isn't being set by the event being sent?

Reply:
I accidentally removed it from the installed method so I just committed a new version with that change.

I didn't remove it from updated because things don't always work the way they're supposed to in ST and calling it from both methods wasn't an issue for the other 9 DTHs that have been approved...

Please let me know if this PR is all set.  Thanks